src/solvers/dgsem_p4est/dg_3d_subcell_limiters.jl

```diff
@@ -0,0 +1,301 @@
+        Ja2_node = get_contravariant_vector(2, contravariant_vectors, i, j, k, element)
+
+        # y direction
+        for jj in (j + 1):nnodes(dg)
+            u_node_jj = get_node_vars(u, equations, dg, i, jj, k, element)
+            # pull the contravariant vectors and compute the average
+            Ja2_node_jj = get_contravariant_vector(2, contravariant_vectors, i, jj, k,
+                                                   element)
+            Ja2_avg = 0.5f0 * (Ja2_node + Ja2_node_jj)
+            # compute the contravariant sharp flux in the direction of the averaged contravariant vector
+            fluxtilde2 = volume_flux(u_node, u_node_jj, Ja2_avg, equations)
+            multiply_add_to_node_vars!(flux_temp, derivative_split[j, jj], fluxtilde2,
+                                       equations, dg, i, j, k)
+            multiply_add_to_node_vars!(flux_temp, derivative_split[jj, j], fluxtilde2,
+                                       equations, dg, i, jj, k)
+        end
+    end
+
+    # FV-form flux `fhat` in y direction
+    fhat2_L[:, :, 1, :] .= zero(eltype(fhat2_L))
+    fhat2_L[:, :, nnodes(dg) + 1, :] .= zero(eltype(fhat2_L))
+    fhat2_R[:, :, 1, :] .= zero(eltype(fhat2_R))
+    fhat2_R[:, :, nnodes(dg) + 1, :] .= zero(eltype(fhat2_R))
+
+    for k in eachnode(dg), j in 1:(nnodes(dg) - 1), i in eachnode(dg),
+        v in eachvariable(equations)
+
+        fhat2_L[v, i, j + 1, k] = fhat2_L[v, i, j, k] +
+                                  weights[j] * flux_temp[v, i, j, k]
+        fhat2_R[v, i, j + 1, k] = fhat2_L[v, i, j + 1, k]
+    end
+
+    # Split form volume flux in orientation 3: z direction
+    flux_temp .= zero(eltype(flux_temp))
+
+    for k in eachnode(dg), j in eachnode(dg), i in eachnode(dg)
+        u_node = get_node_vars(u, equations, dg, i, j, k, element)
+
+        # pull the contravariant vectors in each coordinate direction
+        Ja3_node = get_contravariant_vector(3, contravariant_vectors, i, j, k, element)
+
+        # y direction
+        for kk in (k + 1):nnodes(dg)
+            u_node_kk = get_node_vars(u, equations, dg, i, j, kk, element)
+            # pull the contravariant vectors and compute the average
+            Ja3_node_kk = get_contravariant_vector(3, contravariant_vectors, i, j, kk,
+                                                   element)
+            Ja3_avg = 0.5f0 * (Ja3_node + Ja3_node_kk)
+            # compute the contravariant sharp flux in the direction of the averaged contravariant vector
+            fluxtilde3 = volume_flux(u_node, u_node_kk, Ja3_avg, equations)
+            multiply_add_to_node_vars!(flux_temp, derivative_split[k, kk], fluxtilde3,
+                                       equations, dg, i, j, k)
+            multiply_add_to_node_vars!(flux_temp, derivative_split[kk, k], fluxtilde3,
+                                       equations, dg, i, j, kk)
+        end
+    end
+
+    # FV-form flux `fhat` in y direction
```

Comment:
```suggestion
    # FV-form flux `fhat` in z direction
```

Shouldn't this be `z` direction?